Flaking test TestWriterRedoGC

### Which jobs are flaking?

verify

### Which test(s) are flaking?

TestWriterRedoGC

### Jenkins logs or GitHub Actions link

```log
[2022-06-03T07:49:47.594Z] === Failed

[2022-06-03T07:49:47.594Z] === FAIL: cdc/redo/writer TestWriterRedoGC (0.01s)

[2022-06-03T07:49:47.594Z]     writer_test.go:760: 

[2022-06-03T07:49:47.594Z]         	Error Trace:	writer_test.go:760

[2022-06-03T07:49:47.594Z]         	Error:      	Should have called with given arguments

[2022-06-03T07:49:47.594Z]         	Test:       	TestWriterRedoGC

[2022-06-03T07:49:47.594Z]         	Messages:   	Expected "GC" to have been called with:

[2022-06-03T07:49:47.594Z]         	            	[mock.Anything]

[2022-06-03T07:49:47.594Z]         	            	but actual calls were:

[2022-06-03T07:49:47.594Z]         	            	        []

[2022-06-03T07:49:47.594Z]         	            	[]

[2022-06-03T07:49:47.594Z]         	            	[]

[2022-06-03T07:49:47.594Z]         	            	[]

[2022-06-03T07:49:47.594Z] 

[2022-06-03T07:49:47.594Z] DONE 2495 tests, 1 failure in 527.208s

[2022-06-03T07:49:50.121Z] make: *** [unit_test_in_verify_ci] Error 1

```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

--- a/cdc/redo/writer/writer_test.go
+++ b/cdc/redo/writer/writer_test.go
@@ -36,6 +36,7 @@ import (
 	"github.com/pingcap/tiflow/pkg/uuid"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+	"github.com/uber-go/atomic"
 	"go.uber.org/multierr"
 )
 
@@ -741,8 +742,11 @@ func TestWriterRedoGC(t *testing.T) {
 		mockWriter.On("Close").Return(nil)
 		mockWriter.On("IsRunning").Return(false)
 
+		gcRunCounter := atomic.NewUint32(0)
 		if tt.args.isRunning {
-			mockWriter.On("GC", mock.Anything).Return(nil)
+			mockWriter.On("GC", mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+				gcRunCounter.Inc()
+			})
 		}
 		writer := LogWriter{
 			rowWriter: mockWriter,
@@ -751,7 +755,15 @@ func TestWriterRedoGC(t *testing.T) {
 			cfg:       cfg,
 		}
 		go writer.runGC(context.Background())
-		time.Sleep(time.Duration(defaultGCIntervalInMs+10) * time.Millisecond)
+		if tt.args.isRunning {
+			require.Eventually(t, func() bool {
+				if gcRunCounter.Load() < uint32(2) {
+					return false
+				}
+				return true
+			}, time.Second*1, time.Millisecond,
+				"gc should be called twice, but actual gcRunCounter is: %d", gcRunCounter.Load())
+		}
 
 		writer.Close()
 		mockWriter.AssertNumberOfCalls(t, "Close", 2)

--- a/cdc/redo/writer/writer_test.go
+++ b/cdc/redo/writer/writer_test.go
@@ -751,7 +751,7 @@ func TestWriterRedoGC(t *testing.T) {
 			cfg:       cfg,
 		}
 		go writer.runGC(context.Background())
-		time.Sleep(time.Duration(defaultGCIntervalInMs+1) * time.Millisecond)
+		time.Sleep(time.Duration(defaultGCIntervalInMs+10) * time.Millisecond)
 
 		writer.Close()
 		mockWriter.AssertNumberOfCalls(t, "Close", 2)

